Enable creation of boxing wrappers for some aten operators

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26434 Move Generator ops to c10
* **#26273 Enable creation of boxing wrappers for some aten operators**
* #26272 Add two levels to use_c10_dispatcher
* #26337 c10::KernelFunction

[namedtensor ci]

Differential Revision: [D17393318](https://our.internmc.facebook.com/intern/diff/D17393318/)

smessmer committed:
[wip] Enable creation of boxing wrappers for some aten operators

[namedtensor ci]

Differential Revision: [D17393318](https://our.internmc.facebook.com/intern/diff/D17393318/)

ezyang commented:
For the record: I oppose adding a second partial state here. We should have 100% coverage for boxing wrappers in c10 if we add it. If you can't handle an op, disable it from c10 instead.

smessmer commented:
The reason for the two-step process is that it allows us to delete `globalATenDispatch` earlier. Once all ops are in stage 1, we can already delete it and making them boxed is a problem inside the c10 dispatcher. If we force all ops to immediately be box-able, it will significantly delay the deletion of `globalATenDispatch` and there will be a longer time we have to deal with multiple dispatchers.

smessmer committed:
Update on "[wip] Enable creation of boxing wrappers for some aten operators"


[namedtensor ci]

Differential Revision: [D17393318](https://our.internmc.facebook.com/intern/diff/D17393318/)

[ghstack-poisoned]

ezyang commented:
OK

smessmer committed:
Update on "[wip] Enable creation of boxing wrappers for some aten operators"


[namedtensor ci]

Differential Revision: [D17393318](https://our.internmc.facebook.com/intern/diff/D17393318/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Enable creation of boxing wrappers for some aten operators"


[namedtensor ci]

Differential Revision: [D17393318](https://our.internmc.facebook.com/intern/diff/D17393318/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Enable creation of boxing wrappers for some aten operators"


[namedtensor ci]

Differential Revision: [D17393318](https://our.internmc.facebook.com/intern/diff/D17393318/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Enable creation of boxing wrappers for some aten operators"


[namedtensor ci]

Differential Revision: [D17393318](https://our.internmc.facebook.com/intern/diff/D17393318/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Enable creation of boxing wrappers for some aten operators"


[namedtensor ci]

Differential Revision: [D17393318](https://our.internmc.facebook.com/intern/diff/D17393318/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Enable creation of boxing wrappers for some aten operators"


[namedtensor ci]

Differential Revision: [D17393318](https://our.internmc.facebook.com/intern/diff/D17393318/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Enable creation of boxing wrappers for some aten operators"


[namedtensor ci]

Differential Revision: [D17393318](https://our.internmc.facebook.com/intern/diff/D17393318/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Enable creation of boxing wrappers for some aten operators"


[namedtensor ci]

Differential Revision: [D17393318](https://our.internmc.facebook.com/intern/diff/D17393318/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Enable creation of boxing wrappers for some aten operators"


[namedtensor ci]

Differential Revision: [D17393318](https://our.internmc.facebook.com/intern/diff/D17393318/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Enable creation of boxing wrappers for some aten operators"


[namedtensor ci]

Differential Revision: [D17393318](https://our.internmc.facebook.com/intern/diff/D17393318/)

[ghstack-poisoned]

smessmer committed:
Update on "Enable creation of boxing wrappers for some aten operators"


[namedtensor ci]

Differential Revision: [D17393318](https://our.internmc.facebook.com/intern/diff/D17393318/)

[ghstack-poisoned]

smessmer committed:
Update on "Enable creation of boxing wrappers for some aten operators"


[namedtensor ci]

Differential Revision: [D17393318](https://our.internmc.facebook.com/intern/diff/D17393318/)

[ghstack-poisoned]

smessmer committed:
Update on "Enable creation of boxing wrappers for some aten operators"


[namedtensor ci]

Differential Revision: [D17393318](https://our.internmc.facebook.com/intern/diff/D17393318/)

[ghstack-poisoned]

smessmer committed:
Update on "Enable creation of boxing wrappers for some aten operators"


[namedtensor ci]

Differential Revision: [D17393318](https://our.internmc.facebook.com/intern/diff/D17393318/)

[ghstack-poisoned]

smessmer committed:
Update on "Enable creation of boxing wrappers for some aten operators"


[namedtensor ci]

Differential Revision: [D17393318](https://our.internmc.facebook.com/intern/diff/D17393318/)

[ghstack-poisoned]

smessmer committed:
Update on "Enable creation of boxing wrappers for some aten operators"


[namedtensor ci]

Differential Revision: [D17393318](https://our.internmc.facebook.com/intern/diff/D17393318/)

[ghstack-poisoned]
